src/modules/75-cf/pages/feature-flags-detail/flag-pipeline-tab/__tests__/FlagPipelineTab.test.tsx

```diff
@@ -146,9 +174,184 @@ describe('FlagPipelineTab', () => {
+            name: 'Pipeline 5',
+            description: 'This is a test pipeline'
+          }
+        },
+        loading: false,
+        refetch: getFeaturePipelineRefetchMock
+      } as any)
+    })
+    test('it should update flag pipeline correctly', async () => {
+      renderComponent()
+
+      // click edit button
+      const rbacOptionsMenu = document.querySelector("[data-icon='Options']") as HTMLElement
+      userEvent.click(rbacOptionsMenu)
+      const editButton = screen.getAllByRole('listitem')[0].getElementsByTagName('a')[0]
+      await waitFor(() => {
+        expect(editButton).toHaveTextContent(/edit/)
+      })
+      userEvent.click(editButton)
+
+      await waitFor(() => {
+        expect(screen.getByText('cf.featureFlags.flagPipeline.drawerTitle')).toBeInTheDocument()
+        expect(screen.getAllByRole('listitem')).toHaveLength(mockAvailablePipelines.length)
+      })
+
+      // pipeline should be preselected
+      const preselectedItem = screen.getAllByRole('listitem')[4]
+      await waitFor(() => expect(preselectedItem).toHaveClass('cardActive'))
+
+      // select another pipeline
+      const newSelectedItem = screen.getAllByRole('listitem')[6]
+      userEvent.click(newSelectedItem)
+      await waitFor(() => expect(newSelectedItem).toHaveClass('cardActive'))
+
+      // save
+      const savePipelineButton = screen.getByRole('button', { name: 'cf.featureFlags.flagPipeline.drawerButtonText' })
+      expect(savePipelineButton).toBeInTheDocument()
+
+      userEvent.click(savePipelineButton)
+
       await waitFor(() =>
-        expect(screen.getByText('cf.featureFlags.flagPipeline.noAvailablePipelines')).toBeInTheDocument()
+        expect(patchFeaturePipelineMock).toBeCalledWith({
+          pipelineIdentifier: 'pipeline7',
+          pipelineName: 'Pipeline 7'
+        })
       )
-    }
-  )
+    })
+
+    test('it should delete flag pipeline correctly', async () => {
+      renderComponent()
+
+      // click delete button - assert modal appears
+      const rbacOptionsMenu = document.querySelector("[data-icon='Options']") as HTMLElement
+      userEvent.click(rbacOptionsMenu)
+      const deleteButton = screen.getAllByRole('listitem')[1].getElementsByTagName('a')[0]
+      await waitFor(() => {
+        expect(deleteButton).toHaveTextContent(/delete/)
+      })
+      userEvent.click(deleteButton)
+
+      await waitFor(() => {
+        expect(screen.getByText('cf.featureFlags.flagPipeline.deleteModalTitle')).toBeInTheDocument()
+        expect(screen.getByText('cf.featureFlags.flagPipeline.deleteModalText')).toBeInTheDocument()
+        expect(screen.getByRole('button', { name: 'Confirm' })).toBeInTheDocument()
+        expect(screen.getByRole('button', { name: 'Cancel' })).toBeInTheDocument()
+      })
+
+      // click confirm
+      const confirmButton = screen.getByRole('button', { name: 'Confirm' })
+      expect(confirmButton).toBeInTheDocument()
+      userEvent.click(confirmButton)
```

Comment:
Super minor, but if all you're doing is clicking something, you might as well inline it.
```suggestion
      userEvent.click(screen.getByRole('button', { name: 'Confirm' }))
```

Reply:
fixed